frame/composable-traits/src/staking/mod.rs

```diff
@@ -1,35 +1,138 @@
-use crate::staking::{
-	lock::{Lock, LockConfig},
-	rewards::RewardConfig,
-};
+use crate::staking::lock::{Lock, LockConfig};
 use codec::{Decode, Encode};
 
-use core::fmt::Debug;
-use frame_support::dispatch::DispatchResult;
+use crate::time::DurationSeconds;
+use frame_support::{dispatch::DispatchResult, pallet_prelude::*, BoundedBTreeMap};
 use scale_info::TypeInfo;
+use sp_arithmetic::traits::Zero;
 use sp_runtime::{DispatchError, Perbill, Permill};
 
 pub mod lock;
 pub mod math;
 pub mod nft;
-pub mod rewards;
 
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
-pub struct StakeConfig<DurationPresets, RewardsRate> {
-	pub lock: Option<LockConfig<DurationPresets>>,
-	pub reward: Option<RewardConfig<RewardsRate>>,
+pub type StakingDurationToRewardsMultiplierConfig<Limit> =
+	BoundedBTreeMap<DurationSeconds, Perbill, Limit>;
+
+/// Defines staking duration, rewards and early unstake penalty for a given asset type.
+/// TODO refer to the relevant section in the design doc.
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct Reward<AssetId, Balance> {
+	/// asset id of the reward
+	pub asset_id: AssetId,
+
+	/// Total rewards including inflation for adjusting for new stakers joining the pool. All
+	/// stakers in a pool are eligible to receive a part of this value based on their share of the
+	/// pool.
+	pub total_rewards: Balance,
```

Comment:
this is not part of public API. I think that public API types should reveal only end results, not interals. so this likely to go to staking-rewards/src/types.rs

Reply:
Please split into RewardInput to upserts rewards an RewardState to store at rest.

Reply:
I think that is over refactoring,
1. Get things work according to the design
2. Then refactor if things make sense,

Reply:
please get approve on adding non public stuff into composable traits

Reply:
This is not non-public stuff, it's your assumption

Reply:
i use word public as inter crate and inter pallet usage. modifications provided do make negative unclaimable non existing rewards as part of commont types. while it is just detail of implementation.  do we put internal storage states into composable traits? afaik we do not do this.

Reply:
i suggested tonput these types inside subject pallet as per my comments.

Reply:
so for me it ok if @KaiserKarel will approve this and we will take the direction of put internal pallet details used purely for bookkeeping(these number do not mean anything useful) as part of composable traits.

Reply:
so I think it is weird